### PR TITLE
Download request properties

### PR DESCRIPTION
**Pull request by CraziFuzzy**
Added ability to specify request header properties specific to an individual file download request.  Primarily done to set a specific user-agent that might be required for a given file.

This should close out issue #2.

### CODE REVIEW

File: java/sage/api/Global.java [hunk 1]
```diff
@@ -2171,11 +2190,32 @@ public Object runSafely(Catbert.FastStack stack) throws Exception{
...
+       */
       public Object runSafely(Catbert.FastStack stack) throws Exception{
+        java.util.Properties props = null;
+        if (curNumberOfParameters == 4)
+          props = (java.util.Properties) stack.pop();
         java.io.File dest = getFile(stack);
         String src = getString(stack);
         String serverName = getString(stack);
-        return FileDownloader.getFileDownloader(stack.getUIMgrSafe()).downloadCircularFile(serverName, src, dest);
+        return new FileDownloader(null).downloadFile(serverName, src, dest, true, props);
```
Comment (Narflex) [inline java/sage/api/Global.java]: Why did you change this to construct a new object rather than using the one associated with the UIManager? This method is for foreground downloading which is used during playback, so it needs to use the one linked with that UIManager or it'll break other thing.

Comment (CraziFuzzy) [inline java/sage/api/Global.java]: Honestly, I don't have an answer.  If anything, It looks like a copy/paste error, pulling from the background downloader on accident.  I likely didn't notice the problem because I don't have anything currently that uses the circular download.

Oh, just remembered why I was copying and pasting in the first place.  downloadCircularFile doesn't have an overload that brings in the props, so I went with downloadFile and set the circular boolean to true - but no, it shouldn't be creating a new object, so I'll get that fixed up and make a new pull request.
